Packages/Core/Editor/Drawers/AtomDrawer.cs

```diff
@@ -151,10 +151,21 @@ public override void OnGUI(Rect position, SerializedProperty property, GUIConten
                 {
                     if (GUI.Button(restRect, "Create"))
                     {
-                        drawerData.NameOfNewAtom = "";
                         drawerData.UserClickedToCreateAtom = true;
-
                         EditorGUI.FocusTextInControl(NAMING_FIELD_CONTROL_NAME);
+
+                        if (property.GetParent() is BaseAtom abc)
+                        {
+                            var atomName = AtomNameUtils.CheckForDuplicateAtom(abc.name + AtomNameUtils.CleanPropertyName(property.name)
+                                + AtomNameUtils.FilterLastIndexOf(typeof(T).ToString(), "."));
+                            drawerData.NameOfNewAtom = atomName;
+                        }
+                        else
+                        {
+                            var atomName = AtomNameUtils.CheckForDuplicateAtom(AtomNameUtils.CleanPropertyName(property.name)
+                                + AtomNameUtils.FilterLastIndexOf(typeof(T).ToString(), "."));
+                            drawerData.NameOfNewAtom = atomName;
+                        }
```

Comment:
This could be rewritten with less repetion like this: 
```
                        var baseAtomName = AtomNameUtils.CleanPropertyName(property.name) + AtomNameUtils.FilterLastIndexOf(typeof(T).ToString(), ".");
                        var atomName = property.GetParent() is BaseAtom parentAtom ? parentAtom.name + baseAtomName : baseAtomName;
                        drawerData.NameOfNewAtom = AtomNameUtils.CheckForDuplicateAtom(atomName);
```

Reply:
Fixed in latest commit